src/view/configuration/configuration.jsx

```diff
@@ -528,7 +537,23 @@ const Configuration = () => {
                           </div> */}
 
                           <h3>Context</h3>
-
+                          <div className="u-gapTop">
+                            <InfoTipLayout tip="If you want to add, remove, or modify fields from the event globally, you can configure an `onBeforeEventSend` callback. This callback will be called everytime an event is sent. This callback passes an object with a `xdm` fields. Modify the `xdm` object to change the data that is sent in the event.">
+                              <FieldLabel
+                                labelFor="onBeforeEventSendField"
+                                label="Callback function for modifying data before each event is sent to the server"
+                              />
+                            </InfoTipLayout>
+                            <div>
+                              <WrappedField
+                                id="onBeforeEventSendField"
+                                name={`instances.${index}.onBeforeEventSend`}
+                                component={Textfield}
+                                componentClassName="u-fieldLong"
+                                supportDataElement
+                              />
```

Comment:
@Aaronius What do you think about disabling this textbox?  Do people ever type in the name of the data element?  Disabling the textbox would show that the only thing you can do is click the data element button.

Reply:
Having the button is weird though because it will make two data elements concat-ed together if you click it twice.

Reply:
Yes, customers do sometimes type the name of the data element. I've seen this type of thing come up in Slack conversations with external customers. I would support adding an option to wrappedField that would replace the existing value with the selected data element rather than append, though. I wouldn't want to use it everywhere (sometimes users do something like `%product name% - %product model%`, but in the case of this field (and some others in this extension), it would make sense to just replace the value. Maybe instead of `supportDataElement` create two separate options: `supportAppendDataElement` and `supportReplaceDataElement` or something like that.